src/ansiblelint/rules/jinja.py

```diff
@@ -131,58 +344,175 @@ def test_var_spacing(
+            ),
+            pytest.param(
+                '{{ [ "host", "NA"] }}', '{{ ["host", "NA"] }}', "spacing", id="15"
+            ),
+            pytest.param(
+                "{{ {'dummy_2': {'nested_dummy_1': value_1,\n    'nested_dummy_2': value_2}} |\ncombine(dummy_1) }}",
+                "{{ {'dummy_2': {'nested_dummy_1': value_1,\n    'nested_dummy_2': value_2}} |\ncombine(dummy_1) }}",
+                "spacing",
+                id="17",
+            ),
+            pytest.param("{{ & }}", "", "invalid", id="18"),
+            pytest.param(
+                "{{ good_format }}/\n{{- good_format }}\n{{- good_format -}}\n",
+                "{{ good_format }}/\n{{- good_format }}\n{{- good_format -}}\n",
+                "spacing",
+                id="19",
+            ),
+            pytest.param(
+                "{{ {'a': {'b': 'x', 'c': y}} }}",
+                "{{ {'a': {'b': 'x', 'c': y}} }}",
+                "spacing",
+                id="20",
+            ),
+            pytest.param(
+                "2*(1+(3-1)) is {{ 2 * {{ 1 + {{ 3 - 1 }}}} }}",
+                "2*(1+(3-1)) is {{ 2 * {{1 + {{3 - 1}}}} }}",
+                "spacing",
+                id="21",
+            ),
+            pytest.param(
+                '{{ "absent"\nif (v is version("2.8.0", ">=")\nelse "present" }}',
+                "",
+                "invalid",
+                id="22",
+            ),
+            pytest.param(
+                '{{lookup("x",y+"/foo/"+z+".txt")}}',
+                '{{ lookup("x", y + "/foo/" + z + ".txt") }}',
+                "spacing",
+                id="23",
+            ),
+            pytest.param(
+                "{{ x | map(attribute='value') }}",
+                "{{ x | map(attribute='value') }}",
+                "spacing",
+                id="24",
+            ),
+            pytest.param(
+                "{{ r(a= 1,b= True,c= 0.0,d= '') }}",
+                "{{ r(a=1, b=True, c=0.0, d='') }}",
+                "spacing",
+                id="25",
+            ),
+            pytest.param("{{ r(1,[]) }}", "{{ r(1, []) }}", "spacing", id="26"),
+            pytest.param(
+                "{{ lookup([ddd ]) }}", "{{ lookup([ddd]) }}", "spacing", id="27"
+            ),
+            pytest.param(
+                "{{ [ x ] if x is string else x }}",
+                "{{ [x] if x is string else x }}",
+                "spacing",
+                id="28",
+            ),
+            pytest.param(
+                # "{% if a|int <= 8 -%} iptables {%- else -%} iptables-nft {%- endif %}",
+                "{% if a|int <= 8 -%} iptables {%- else -%} iptables-nft {%- endif %}",
+                "{% if a | int <= 8 -%} iptables {%- else -%} iptables-nft {%- endif %}",
+                "spacing",
+                id="29",
+            ),
+            pytest.param(
+                # "- 2" -> "-2", minus does not get separated when there is no left side
+                "{{ - 2 }}",
```

Comment:
The AST calls this an Unary node. I looked through the parser, and I don't see an easy way to identify this based only on the lexer tokens.

So maybe track the previous non-whitespace token. See if the token before the `-` or `+` operator is also an operator (one of https://github.com/pallets/jinja/blob/main/src/jinja2/lexer.py#L112-L139 except for right tokens: `]`, `)`, `}`) or if the previous token is a variable start token `{{`.